src/lib.rs

```diff
@@ -25,37 +25,70 @@
 //!     docmatic::assert_file("README.md");
 //! }
 //! ```
+//!
+//! Or, to pass additional arguments to `rustdoc`:
+//!
+//! ```rust
+//! extern crate docmatic;
+//!
+//! fn test_readme() {
+//!     let args = [("--cfg", "test"), ("--cfg", "rustdoc")];
+//!     docmatic::assert_file_with("README.md", &args);
+//! }
+//! ```
 
-extern crate glob;
 extern crate which;
 
-use std::path;
+use std::path::Path;
+use std::ffi::OsStr;
 
 pub fn assert_file<P>(documentation: P)
 where
-    P: AsRef<path::Path>,
+    P: AsRef<Path>,
 {
-    assert_file_impl(documentation.as_ref())
+    assert_file_impl::<&str>(documentation.as_ref(), &Vec::new())
 }
 
-fn assert_file_impl(documentation: &path::Path) {
+pub fn assert_file_with<P, S>(documentation: P, args: &[S])
+where
+    P: AsRef<Path>,
+    S: AsRef<OsStr>,
+{
+    assert_file_impl(documentation.as_ref(), args)
+}
+
+fn assert_file_impl<S>(documentation: &Path, args: &[S])
+where
+    S: AsRef<OsStr>,
+{
     let rustdoc = which::which("rustdoc").expect("run with rust toolchain available");
 
-    let current_exe = std::path::Path::new(&std::env::current_exe().unwrap())
-        .canonicalize()
+    let current_exe = std::env::current_exe()
+        .and_then(|ref p| p.canonicalize())
         .unwrap();
     let deps = current_exe.parent().unwrap();
 
     let mut cmd = std::process::Command::new(rustdoc);
     cmd.arg("--verbose")
-        .args(&["--library-path", deps.to_str().unwrap()])
+        .arg("--library-path")
+        .arg(deps)
         .arg("--test")
         .arg(documentation);
 
+    for arg in args.iter() {
```

Comment:
My thought for how to handle this case was to provide an API to wrap rustdoc rather than having people directly call rustdoc.

For example, we might have a builder that looks like
```rust
docmatic::Assert::new("README.md")
    .library_path(path)
    .unwrap()
```
or
```rust
docmatic::Assert::new()
    .library_path(path)
    .file("README.md")
```

Reply:
I will say, if this seems like too much work at the moment, I can understand us leaving it as-is for now and then I can later go back in and refactor it.

Another option is we could mark the `with` variant as `unsafe` or hidden behind an `unstable` feature.